### PR TITLE
feat: jrpc for get all txs

### PR DESCRIPTION
**Pull request by Cifko**
Description
---
Add jrpc call the get all transactions, either by status or all at once.

Motivation and Context
---
We need this for the UI.

How Has This Been Tested?
---
Manually.

What process can a PR reviewer use to test or verify this change?
---
Call the new jrpc method on a wallet with some transactions.

Breaking Changes
---

- [x] None
- [ ] Requires data directory to be deleted
- [ ] Other - Please specify

### CODE REVIEW

**Review by sdbondi — DISMISSED** [review]

File: applications/tari_dan_wallet_daemon/src/handlers/transaction.rs [hunk 1]
```diff
@@ -184,6 +186,27 @@ pub async fn handle_get(
     })
 }
 
+pub async fn handle_get_all_by_status(
```
Comment (jorgeantonio21) [inline applications/tari_dan_wallet_daemon/src/handlers/transaction.rs:189]: I would suggest rename it to `handle_get_all_txs_by_status`

File: applications/tari_dan_wallet_daemon/src/jrpc_server.rs [hunk 2]
```diff
@@ -104,6 +104,7 @@ async fn handler(
             "get" => call_handler(context, value, token, transaction::handle_get).await,
             "get_result" => call_handler(context, value, token, transaction::handle_get_result).await,
             "wait_result" => call_handler(context, value, token, transaction::handle_wait_result).await,
+            "get_all_by_status" => call_handler(context, value, token, transaction::handle_get_all_by_status).await,
```
Comment (jorgeantonio21) [inline applications/tari_dan_wallet_daemon/src/jrpc_server.rs:107]: Could we rename it to `get_all_txs_by_status`? Or `get_all_transacations_by_status`.

File: clients/wallet_daemon_client/src/types.rs [hunk 3]
```diff
@@ -108,6 +108,11 @@ pub struct TransactionGetRequest {
     pub hash: FixedHash,
 }
 
+#[derive(Debug, Clone, Deserialize, Serialize)]
+pub struct TransactionGetAllRequest {
```
Comment (jorgeantonio21) [inline clients/wallet_daemon_client/src/types.rs:112]: Can we rename the struct to `GetAllTransactionsRequest` or `GetAllTransactionsByStatusRequest` ?  I think we use `Get` at the beginning in the previous examples.

File: clients/wallet_daemon_client/src/types.rs [hunk 4]
```diff
@@ -124,6 +129,16 @@ pub struct TransactionGetResponse {
     pub transaction_failure: Option<RejectReason>,
 }
 
+#[derive(Debug, Clone, Deserialize, Serialize)]
+pub struct TransactionGetAllResponse {
```
Comment (jorgeantonio21) [inline clients/wallet_daemon_client/src/types.rs:133]: Can we rename it to `GetAllTransactionsResponse` or to `GetAllTransactionsByStatusResponse` ?

**Review by jorgeantonio21 — APPROVED** [review]
LGTM !

nit: I proposed to do some renaming, that I think it helps a lot on the reading.

File: applications/tari_dan_wallet_daemon/src/jrpc_server.rs (hunk 2 above)
Comment (Cifko) [reply applications/tari_dan_wallet_daemon/src/jrpc_server.rs:107]: it's a `transacation.get_all_by_status` command, that sounds redundant to me and also not consistent with other transactions commands.

File: applications/tari_dan_wallet_daemon/src/handlers/transaction.rs (hunk 1 above)
Comment (Cifko) [reply applications/tari_dan_wallet_daemon/src/handlers/transaction.rs:189]: Redundant and not consistent with rest of the file. e.g. just after this is `handle_get_result`

File: clients/wallet_daemon_client/src/types.rs (hunk 3 above)
Comment (Cifko) [reply clients/wallet_daemon_client/src/types.rs:112]: I want to keep it consistent with `TransactionGetRequest` `TransactionSubmitRequest` etc.

File: clients/wallet_daemon_client/src/types.rs (hunk 4 above)
Comment (Cifko) [reply clients/wallet_daemon_client/src/types.rs:133]: Same :-)

**Review by sdbondi — APPROVED** [review]

**Pull request merged**